sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ComplexSearchField.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A helper Field model to build a simple search field.
+ */
+public class ComplexSearchField {
+    private String name;
+    private ComplexDataType dataType;
+    private List<Field> fieldList;
+
+    public String getName() {
+        return name;
+    }
+
+    public ComplexSearchField setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    public ComplexDataType getDataType() {
+        return dataType;
+    }
+
+    public ComplexSearchField setDataType(ComplexDataType dataType) {
+        this.dataType = dataType;
+        return this;
+    }
+
+    public List<Field> getFieldList() {
+        return fieldList;
+    }
+
+    public ComplexSearchField setSimpleFieldList(List<SimpleSearchField> fieldList) {
```

Comment:
Why the suffixes with "List"? I think it's implied by the parameter name. Same with `getFieldList`. I think:

```java
public List<Field> getFields() {}
```

Reply:
> Is there a reason this can't be done in Java? It looks like some normal reflection where you probe the properties for a JSON object and then set it.

Sorry, which part cannot be done in Java? The major purpose of the new field type is to help user build correct field (as much as possible) in local before sending out to the service.

The structure here is not all scenarios we have to cover. Will collect more points from .Net and service team.

Reply:
Heath's comment suggests that the behaviour in .NET is what this code should look like.
https://github.com/Azure/azure-sdk-for-java/pull/10309#pullrequestreview-395132635

Reply:
Thanks!

Will switch the field builder for taking the search document type.